File: VB/Applications/AvaloniaLog4NetVBDI/ViewModels/MainViewModel.vb
Imports LogViewerVB.Core.ViewModels

Public Class MainViewModel : Inherits ViewModelBase

#Region "Constructor"

    Public Sub New(logViewer As LogViewerControlViewModel)
        Me.LogViewer = logViewer
    End Sub

#End Region

#Region "Properties"

    Public Property LogViewer As LogViewerControlViewModel

#End Region

End Class

File: CSharp/Applications/AvaloniaLoggingDI/ViewModels/MainViewModel.cs
using LogViewer.Core.ViewModels;

namespace AvaloniaLoggingDI.ViewModels;

public class MainViewModel : ViewModelBase
{
    #region Constructor

    public MainViewModel(LogViewerControlViewModel logViewer)
    {
        LogViewer = logViewer;
    }

    #endregion

    #region Properties

    public LogViewerControlViewModel LogViewer { get; }

    #endregion
}